### PR TITLE
DM-39670: fix running the CSC in simulation mode

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/atdome/dome_csc.py [hunk 1]
```diff
@@ -566,34 +545,26 @@ async def connect(self):
...
+                    self.client.start_task, timeout=self.config.connection_timeout
                 )
+                print("Connected; wait for prompt")
                 # drop welcome message
                 await asyncio.wait_for(
-                    self.reader.readuntil(">".encode()),
-                    timeout=self.config.read_timeout,
+                    self.client.readuntil(b">"), timeout=self.config.read_timeout
                 )
+            print("Fully connected")
```
Comment (wvreeven) [inline python/lsst/ts/atdome/dome_csc.py:558]: Is this print necessary? There is a debug log statement in the next line. Maybe change that to info instead if you really want output?

File: python/lsst/ts/atdome/dome_csc.py [hunk 2]
```diff
@@ -566,34 +545,26 @@ async def connect(self):
...
             self.log.debug("connected")
         except Exception as e:
-            err_msg = (
-                f"Could not open connection to host={host}, port={self.config.port}"
-            )
+            err_msg = f"Could not open connection to host={host}, port={self.config.port}: {e!r}"
```
Comment (wvreeven) [inline python/lsst/ts/atdome/dome_csc.py:561]: `port={self.config.port}` -> `port={port}`? In lines 545 and 547 you set that local variable.

**Review by wvreeven — APPROVED** [review]
Looks good in general. I left a few comments for your consideration.

File: python/lsst/ts/atdome/dome_csc.py (hunk 1 above)
Comment (r-owen) [reply python/lsst/ts/atdome/dome_csc.py:558]: Oops. Not necessary -- it was debugging code. I'll take it out and look for other print statements.

File: python/lsst/ts/atdome/dome_csc.py (hunk 2 above)
Comment (r-owen) [reply python/lsst/ts/atdome/dome_csc.py]: Good catch. Also I should set port and host at the same point in the code and stop using salobj.LOCAL_HOST.

File: conda/meta.yaml [hunk 3]
```diff
@@ -8,18 +8,7 @@ source:
...
-  script_env:
-    - PATH
-    - PYTHONPATH
-    - LD_LIBRARY_PATH
-    - OSPL_HOME
-    - PYTHON_BUILD_VERSION
-    - PYTHON_BUILD_LOCATION
-    - TS_CONFIG_ATTCS_DIR
-    - LSST_DDS_DOMAIN
+  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed .
```
[the remark this answers is not in the document]
Comment (r-owen) [reply conda/meta.yaml:11]: Thank you for catching that. I also sorted the entries in each subsection.

**Pull request merged**